fix: add parameters guide with example (#339)
screwdriver-cd/screwdriver-cd.github.io: docs/user-guide/configuration/parameters.md

---
layout: main
title: Parameters
category: User Guide
menu: menu
toc:
    - title: Parameters
      url: "#definition"
      active: true
---
## Parameters Definition
There are 2 ways of defining parameters, see:

```yaml

> paramters:

Reviewer: ```suggestion
parameters:
```

    nameA: "value1"
    nameB:
        value: "value2"
        description: "description of nameB"
```

**Parameters** is a dictionary which expects `key:value` pairs.

```yaml

> nameA: "value1"

Reviewer: ```suggestion
parameters:
    nameA: "value1"
```

I think it's better to have full example so people have context

```

>  `key: string` is a shorthand for writting as `key:value`

Reviewer: ```suggestion
`key: string` (see example above) is a shorthand for writting as `key:value` (see example below).
```

```yaml

> nameA:

Reviewer: ```suggestion
parameters:
    nameA:
```
same reason as above

    value: "value1"
    description: ""
```

> These two are identical with description to be an empty string

Reviewer: ```suggestion
These two examples above are equivalent.
```

## Example

> ```yaml

Reviewer: ```suggestion
You can see a full screwdriver.yaml example below:
```yaml
```

shared:
    image: node:8

parameters:
    region: "us-west-1"
    az:
        value: "1"
        description: "default availability zone"

jobs:
    main:
        requires: [~pr, ~commit]
        steps:
            - step1: 'echo "Region: $(meta get parameters.region.value)"'
            - step2: 'echo "AZ: $(meta get parameters.az.value)"'
```

> You can also preview the parameters that being used during a build in `Setup` -> `sd-setup-init` step

Reviewer: ```suggestion
You can also preview the parameters that are used during a build in the `Setup` -> `sd-setup-init` step.
```

Pipeline Preview Screenshot:

![image](../assets/parameters1-event-start.png)

![image](../assets/parameters2-sd-init-step.png)

![image](../assets/parameters3-event-view.png)

> Please see [parameters-build-sample](https://github.com/screwdriver-cd-test/parameters-build-sample) as example.

Reviewer: ```suggestion
Please see the [parameters-build-sample](https://github.com/screwdriver-cd-test/parameters-build-sample) as an example.
```

---
layout: main
title: Parameters
category: User Guide
menu: menu
toc:
    - title: Parameters

>       url: "#definition"

Reviewer: ```suggestion
      url: "#defining-parameters"
```

      active: true
---

> ## Parameters Definition

Reviewer: ```suggestion
## Defining Parameters
```

> There are 2 ways of defining parameters, see:

Reviewer: ```suggestion
There are 2 ways to define parameters, you can see them both in the example below.
```
